bedshift/bedshift.py

```diff
@@ -247,16 +246,24 @@ def shift_from_file(self, fp, shiftrate, shiftmean, shiftstdev, delimiter="\t"):
         num_shift = int(rows * shiftrate)
 
         intersect_regions = self._find_overlap(fp)
-        interlen = len(intersect_regions)
+        rows2shift = intersect_regions.sample(frac=shiftrate)
+        indices2shift = []
+        for _, r in rows2shift.iterrows():
+            chrom = r[0]
+            start = r[1]
+            end = r[2]
+            index = (self.bed[(self.bed[0]  == chrom) & (self.bed[1] == start)].index.tolist())
+            
+            if len(index) > 0:
+                indices2shift.append(index[0])
+        
+        interlen = len(indices2shift)
         if num_shift > interlen:
-            _LOGGER.warning(
-                "Number of regions to be shifted ({}) is larger than the provided bedfile size ({}). Shifting {} regions.".format(
-                    num_shift, interlen, interlen
-                )
-            )
-            num_shift = interlen
-        rows2shift = random.sample(list(range(interlen)), num_shift)
-        return self.shift(shiftrate, shiftmean, shiftstdev, rows2shift)
+            _LOGGER.warning("Number of regions to be shifted ({}) is larger than the desired number of regions to be shifted: ({}). Shifting {} regions.".format(num_shift, interlen, interlen))
```

Comment:
This new warning message doesn't make sense. It should be more like "Desired regions shifted {} is greater than the number of overlaps found {}."

Also, if `indices2shift` is larger than `num_shift`, it should be downsampled so we are not shifting too many rows.